libc-test/build.rs

```diff
@@ -350,6 +353,9 @@ fn main() {
     cfg.skip_fn(move |name| {
         // skip those that are manually verified
         match name {
+            // glibc-2.24 deprecates these functions
+            "readdir_r" | "readdir64_r" => true,
```

Comment:
The error message is pasted below.
```
out/all.c: In function ‘__test_fn_readdir_r’:
out/all.c:2061:17: error: ‘readdir_r’ is deprecated [-Werror=deprecated-declarations]
                 return readdir_r;
                 ^~~~~~
out/all.c:16:0:
/usr/include/dirent.h:183:12: note: declared here
 extern int readdir_r (DIR *__restrict __dirp,
            ^~~~~~~~~
```
Perhaps we can pass `-Wno-deprecated-declarations` to the compiler? Is that acceptable?